api/app/services/rag/vocabulary_manager.py

```
@@ -0,0 +1,284 @@
+
+            try:
+                # Use file locking for concurrent read safety
+                with portalocker.Lock(self.vocab_path, "r", timeout=10) as f:
+                    vocab_json = f.read()
+
+                tokenizer.load_vocabulary(vocab_json)
+                logger.info(
+                    f"Loaded vocabulary from {self.vocab_path} "
+                    f"({tokenizer.vocabulary_size} tokens)"
+                )
+                return True
```

Ruff (0.14.14)

Consider moving this statement to an `else` block

(TRY300)